docs/ops/sequence/CTCLoss_4.md

```diff
@@ -0,0 +1,125 @@
+Compute probabilities of these aligned paths (alignments) as follows:
+\f[
+p(S) = \prod_{t=1}^{L_i} p_{t,i,ct}
+\f]
+
+3. Finally, compute negative sum of log-probabilities of all found alignments:
+\f[
+CTCLoss = \minus \sum_{S} \ln p(S)
+\f]
+
+**Note**:  This calculation scheme does not provide steps for optimal implementation and primarily serves for better explanation.
+
+**Attributes**
+
+* *preprocess_collapse_repeated*
+
+  * **Description**: *preprocess_collapse_repeated* is a flag for a preprocessing step before loss calculation, wherein repeated labels in `labels[i,:]` passed to the loss are merged into single labels.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+* *ctc_merge_repeated*
+
+  * **Description**: *ctc_merge_repeated* is a flag for merging repeated characters in a potential alignment during the CTC loss calculation.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: True
+  * **Required**: *no*
+
+* *unique*
+
+  * **Description**: *unique* is a flag to find unique elements for a target `labels[i,:]` before matching with potential alignments. Unique elements in the processed `labels[i,:]` are sorted in the order of their occurence in original `labels[i,:]`. For example, the processed sequence for `labels[i,:]=(0,1,1,0,1,3,3,2,2,3)` of length `label_length[i]=10` will be `(0,1,3,2)` in case *unique* equal to True.
+  * **Range of values**: True or False
+  * **Type**: `boolean`
+  * **Default value**: False
+  * **Required**: *no*
+
+**Inputs**
+
+* **1**: `logits` - Input tensor with a batch of sequences of logits. Type of elements is *T_F*. Shape of the tensor is `[N, T, C]`, where `N` is the batch size, `T` is the maximum sequence length and `C` is the number of classes including the blank. Required.
+
+* **2**: `logit_length` - 1D input tensor of type *T_IND* and of a shape `[N]`. The tensor must consist of non-negative values not greater than `T`. Lengths of input sequences of logits `logits[i,:,:]`. Required.
+
+* **3**: `labels` - 2D tensor with shape `[N, T]` of type *T_IND*. A length of a target sequence `labels[i,:]` is equal to `label_length[i]` and must contain of integers from a range `[0; C-1]` except `blank_index`. Required.
+
+* **4**: `label_length` - 1D tensor of type *T_IND* and of a shape `[N]`. The tensor must consist of non-negative values not greater than `T` and `label_length[i] <= logit_length[i]` for all possible `i`.  Required.
+
+* **5**: `blank_index` - Scalar of type *T_IND*. Set the class index to use for the blank label. Default value is `C-1`. Optional.
+
+**Output**
+
+* **1**: Output tensor with shape `[N]`, negative sum of log-probabilities of alignments. Type of elements is *T_F*.
+
+**Types**
+
+* *T_F*: any supported floating point type.
+
+* *T_IND*: any supported integer type.
```

Comment:
I am not sure if "any" integer type makes sense. For example u1 is clearly insufficient for this op. I think fro other ops we use either I32 or I64 for data such as indices or lengths. Also as it is written now this op has 4 integer inputs of same integer type - is this intentional?

Reply:
Good catch! Let's use `int32` and `int64` for them. Now we use two different types for indices and length those make more sense to me. TF and Pytorch specs does say nothing about types.